src/app/account-pets/account-pets.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+
+import { Pet } from '../pet';
+import { PetService } from '../pet.service';
+
+@Component({
+  selector: 'app-account-pets',
+  templateUrl: './account-pets.component.html',
+  styleUrls: ['./account-pets.component.css']
+})
+export class AccountPetsComponent implements OnInit {
+
+  pets: Pet[] = [];
+
+  constructor(private petService: PetService) { }
+
+  ngOnInit(): void {
+    this.getPets();
+  }
+
+  getPets(): void {
+    this.petService.getPets().subscribe(pets => this.pets = pets);
```

Comment:
When you subscribe to an observable, that subscription lives for as long as you specify. If you don't specify, it lives for the lifetime of the application, even if you leave the page. In this case you probably just want to set this.pets equal to the observable and not subscribe to it. We can talk about this pattern and maybe I can show it.